### PR TITLE
Refactor deletion

### PR DESCRIPTION
**Pull request by delvh**
This PR introduces new generic deletion methods
- `func DeleteByID[T any](ctx context.Context, id int64) (int64, error)`
- `func DeleteByIDs[T any](ctx context.Context, ids ...int64) error`
- `func Delete[T any](ctx context.Context, opts FindOptions) (int64, error)`

So, we no longer need any specific deletion methods and can just use those instead.

Replacement of #28450

Closes #28450

### CODE REVIEW

File: models/db/context.go [hunk 1]
```diff
@@ -213,16 +213,36 @@ func ExistByID[T any](ctx context.Context, id int64) (bool, error) {
...
+		return nil
+	}
+
+	var bean T
+	_, err := GetEngine(ctx).In("id", ids).NoAutoCondition().NoAutoTime().Delete(&bean)
+	return err
+}
+
+func Delete[T any](ctx context.Context, opts FindOptions) (int64, error) {
+	if opts == nil {
```
Comment (lunny) [inline models/db/context.go]:
```suggestion
	if opts == nil  || !opts.ToConds().IsValid() {
```

File: models/repo/archiver.go [hunk 2]
```diff
@@ -114,6 +100,7 @@ func UpdateRepoArchiverStatus(ctx context.Context, archiver *RepoArchiver) error
 
 // DeleteAllRepoArchives deletes all repo archives records
 func DeleteAllRepoArchives(ctx context.Context) error {
+	// 1=1 to enforce delete all data, otherwise it will delete nothing
 	_, err := db.GetEngine(ctx).Where("1=1").Delete(new(RepoArchiver))
```
Comment (lunny) [inline models/repo/archiver.go:104]: Maybe we should have a new function in `db/context.go` Like `DeleteAllRecordsOnTable`.

Comment (delvh) [reply models/repo/archiver.go:104]: But not in this PR

**Review by lunny — APPROVED** [review]

**Review by wxiaoguang — APPROVED** [review]
Many features should be better to be provided by ORM framework

**Review by nanguanlin6 — APPROVED** [review]

**Pull request merged**